### DISCUSSION TITLE
[Chapter 08] extends JpaRepository<?,?>

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#110] [Chapter 08. Data Persistence] [answered]
<img width="744" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/71637466/8d9c59e2-a429-4fc0-a360-34b3a6cd7c0c">

### Q1.
> 책 예제 코드에서 JpaRepository 인터페이스를 상속한 HotelRepository 코드가 나온다.
여기서 `@Repository` 를 생략해도 잘 실행이 되는데 왜 그런 것일까? 

### Q2.
<img width="844" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/71637466/3b90407d-132a-4fb9-a02a-8fb4af3d0257">

> 책 예제 코드에서 `@TestPropertySource` 을 사용하여 yaml 을 설정하는 코드가 나오는데 확인한 결과 `application-test.yaml` 이 적용 안되는 현상이 보인다. 왜 그런것일까?

**Comment by ahngilwoong** [comment, marked as answer]
Q1. JpaRepository의 구현체인 SimpleJpaRepository.class에
이미 Repository 어노테이션이 적용되어 있어
컴포넌트 스캔의 대상이 되기 때문입니다.

@Repository는 Component 어노테이션에 더불어 DB연동중 발생하는 에러를 스프링 에러로 바꿔주는 역할을 한다고 합니다 !


Q2. 잘 모르겠어서 다른 분들의 답변 부탁드립니다 ! (혹시 확장자 yaml이 아니라 yml ..?)

**Reply by corock** [reply]
cf. https://brunch.co.kr/@anonymdevoo/40

**Reply by ahngilwoong** [reply]
[repository 추가글1](https://starryeye.tistory.com/97)
[repository 추가글2](https://www.google.com/search?q=%40repository+dataaccessexcpetion&oq=%40repository+dataaccessexcpetion&gs_lcrp=EgZjaHJvbWUyBggAEEUYOdIBCDk2MjRqMGoxqAIAsAIA&sourceid=chrome&ie=UTF-8#ip=1)

**Comment by JoisFe** [comment]
@coalong 님 파일 구조입니다.

<img width="336" alt="image" src="https://github.com/cowaine/spring-boot-for-msa/assets/90208100/cb1a1016-e89d-42d3-8682-df77ec1231cd">

**Comment by Irisation23** [comment]
2. 답변을 첨부해보면.

```
@PropertySource 는 기본적으로 yaml, yml을 로드하지 않는다고 합니다.
```

아래의 링크를 참조해 보시면 해결방법을 강구할 수 있을 것 같습니다.

https://pooney.tistory.com/88

**Comment by Youngju-Jang** [comment]
@TestPropertySource(properties = {"spring.config.location = classpath:application-test.yaml"})

이거 저번에 지호님이 올려주신대로 설정하니까 되더라구요

**Reply by corock** [reply]
cf. https://corono.tistory.com/31

**Comment by coalong** [comment]
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config

**Comment by Irisation23** [comment]
### 금주 간단 정리 Check

1. `on` ~~ 했을때 (특정 이벤트를 감지 했을때) 라는 키워드를 내포할 수 있음
2. properites 설정을 기반으로 한다면 2번 질문에 해답을 구할 수 있음

**Comment by JoisFe** [comment]
책은 .properties 입니다.
